lib/Sema/TypeCheckType.cpp

```diff
@@ -565,9 +565,7 @@ Type TypeChecker::resolveTypeInContext(TypeDecl *typeDecl, DeclContext *foundDC,
 
     if (selfType->is<GenericTypeParamType>()) {
       if (typeDecl->getDeclContext()->getSelfProtocolDecl()) {
-        if (isa<AssociatedTypeDecl>(typeDecl) ||
-            (isa<TypeAliasDecl>(typeDecl) &&
-             !cast<TypeAliasDecl>(typeDecl)->isGeneric())) {
+        if (isa<AssociatedTypeDecl>(typeDecl)) {
```

Comment:
Maybe we can land this change separately. Is there a test case for it?

Reply:
We can, there are test cases for `associatedtype` preference over `typealias` in the suite, but the [test case](https://github.com/apple/swift/pull/23489#issuecomment-548614488) related to the crash relies on this proposal.

Reply:
It happens this fixes https://github.com/apple/swift/pull/27987 too. Going to land this separately now that we have an independent test case.